src/Data/Entity.php

```diff
@@ -13,25 +17,63 @@
 
     public function __construct($data = [])
     {
-        $this->data = $data;
+        $this->mapData($data);
     }
 
     public function getId()
     {
-        return $this->id;
+        return (int) $this->id;
+    }
+
+    public function has($field)
+    {
+        return array_key_exists($field, $this->properties) || in_array($field, $this->properties);
+    }
+
+    public function getDate($field)
+    {
+        return new Carbon($this->{$field});
+    }
+
+    public function getData()
+    {
+        $object = [];
+        foreach ($this->properties as $k=>$v) {
+            if (is_numeric($k)) {
+                $k = $v;
+                $v = null;
+            }
+            $object[$k] = $this->{$k};
+        }
+
+        return $object;
     }
 
     public function mapData(array $row)
     {
-        $this->data = $row;
+        foreach ($this->properties as $k=>$v) {
+            if (is_numeric($k)) {
+                $k = $v;
+                $v = null;
+            }
+            if (!array_key_exists($k, $row)) {
+                continue;
+            }
```

Comment:
The idea behind them was to allow for us to define properties so we can have the name of the field be the value, or the key.
```php
$p = [
    'id',
    'created_at' => ['type' => 'datetime'],
];
```

It's mostly a hack to get something to work. All this is in flux, and I'm fine with it changing significantly.